library/powersync/AttachmentQueue.ts

```diff
@@ -0,0 +1,406 @@
+      cacheLimit: options.cacheLimit ?? ATTACHMENT_CACHE_LIMIT
+    };
+    this.downloadQueue = new Set<string>();
+  }
+
+  protected get powersync() {
+    return this.options.powersync;
+  }
+
+  protected get storage() {
+    return this.options.storage;
+  }
+
+  get table() {
+    return ATTACHMENT_TABLE;
+  }
+
+  async init() {
+    // Ensure attachment directory exists
+    await this.storage.makeDir(ATTACHMENT_STORAGE_PATH);
+
+    const attachmentIds = (await this.options.getAttachmentIds?.()) || [];
+    if (attachmentIds.length > 0) {
+      await this.markForSync(attachmentIds);
+    }
+
+    this.watchUploads();
+    this.watchDownloads();
+
+    setInterval(() => this.trigger(), ATTACHMENT_QUEUE_INTERVAL);
+  }
+
+  trigger() {
+    this.uploadRecords();
+    this.downloadRecords();
+    this.expireCache();
+  }
+
+  async getAttachmentRecord(id: string): Promise<AttachmentRecord | null> {
+    return this.powersync.getOptional<AttachmentRecord>(`SELECT * FROM ${this.table} WHERE id = ?`, [id]);
+  }
+
+  async savePhoto(base64Data: string): Promise<AttachmentRecord> {
+    const photoAttachment = this.newPhotoRecord();
+    photoAttachment.local_uri = this.getLocalUri(photoAttachment.filename);
+    await this.storage.writeFile(photoAttachment.local_uri!, base64Data, { encoding: FileSystem.EncodingType.Base64 });
+
+    const fileInfo = await FileSystem.getInfoAsync(photoAttachment.local_uri!);
+    if (fileInfo.exists) {
+      photoAttachment.size = fileInfo.size;
+    }
+
+    return this.saveToQueue(photoAttachment);
+  }
+
+  async saveToQueue(record: Omit<AttachmentRecord, 'timestamp'>): Promise<AttachmentRecord> {
+    const updatedRecord: AttachmentRecord = {
+      ...record,
+      timestamp: new Date().getTime()
+    };
+
+    await this.powersync.execute(
+      `INSERT OR REPLACE INTO ${this.table} (id, timestamp, filename, local_uri, media_type, size, state) VALUES (?, ?, ?, ?, ?, ?, ?)`,
+      [
+        updatedRecord.id,
+        updatedRecord.timestamp,
+        updatedRecord.filename,
+        updatedRecord.local_uri || null,
+        updatedRecord.media_type || null,
+        updatedRecord.size || null,
+        updatedRecord.state
+      ]
+    );
+
+    return updatedRecord;
+  }
+
+  async update(record: Omit<AttachmentRecord, 'timestamp'>): Promise<void> {
+    const timestamp = new Date().getTime();
+    await this.powersync.execute(
+      `UPDATE ${this.table}
+             SET 
+                timestamp = ?,
+                filename = ?,
+                local_uri = ?,
+                size = ?,
+                media_type = ?,
+                state = ?
+             WHERE id = ?`,
+      [timestamp, record.filename, record.local_uri || null, record.size, record.media_type, record.state, record.id]
+    );
+  }
+
+  async markForSync(ids: string[]): Promise<void> {
+    if (ids.length > 0) {
+      // Mark attachments for sync
+      await this.powersync.execute(
```

Comment:
Shouldn't this also handle creating new attachment rows? And potentially "unmarking" attachments for sync when not in the `ids` array anymore?